src/Orleans.Core/Timers/TimerManager.cs

```diff
@@ -0,0 +1,354 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.Threading;
+
+namespace Orleans.Timers
+{
+    public static class TimerManager
```

Comment:
Why static?  I thought we were actively moving away from the use of statics?

Reply:
Similar to Task.Delay design. Timer management is naturally static and we'd need to rework a huge amount of code to inject some kind of `ITimerManager`

Reply:
I don't think we use task.delay a lot outside of test.

How about a compromise, we keep implementations static and internal but wrap it in an interface.  That way we can address this later without changes to the public surface.  Once we expose this new feature as part of our public surface as a static, our options become more limited.